heron/api/tests/java/com/twitter/heron/streamlet/impl/StreamletImplTest.java

```diff
@@ -343,6 +344,60 @@ public void testConfigBuilder() {
     assertEquals(multiSetConfig.getSerializer(), Config.Serializer.KRYO);
   }
 
+  @Test
+  public void testDefaultStreamletNameIfNotSet() {
```

Comment:
In general, I think that the changes suggested in this PR are already covered in #2559. But I do think these tests would be nice to have. Could you possibly re-submit this or a different PR with just the tests once #2559 is merged?